runtime/interpreter/interpreter.go

```diff
@@ -3535,6 +3609,36 @@ var runtimeTypeConstructors = []runtimeTypeConstructor{
 			},
 		),
 	},
+	{
+		name: "InclusiveRangeType",
+		converter: NewUnmeteredHostFunctionValue(
+			sema.InclusiveRangeTypeFunctionType,
+			func(invocation Invocation) Value {
+				typeValue, ok := invocation.Arguments[0].(TypeValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
+
+				ty := typeValue.Type
+				// InclusiveRanges must hold integers
+				elemSemaTy := invocation.Interpreter.MustConvertStaticToSemaType(ty)
+				if !elemSemaTy.Tag().BelongsTo(sema.IntegerTypeTag) {
+					return Nil
+				}
+
+				return NewSomeValueNonCopying(
+					invocation.Interpreter,
+					NewTypeValue(
+						invocation.Interpreter,
+						NewInclusiveRangeStaticType(
+							invocation.Interpreter,
+							ty,
+						),
+					),
+				)
```

Comment:
Can you add a test to cover these lines? Creating a type-value in cadence should cover this path.

Reply:
Added `TestInterpretInclusiveRangeType` and `TestCheckInclusiveRangeTypeConstructor`.